scripts/Diffraction/isis_powder/osiris.py

```diff
@@ -0,0 +1,185 @@
+        """
+        run_number_string_key = self._generate_run_details_fingerprint(run_number_string,
+                                                                       self._inst_settings.file_extension)
+
+        if run_number_string_key in self._run_details_cached_obj:
+            return self._run_details_cached_obj[run_number_string_key]
+
+        self._run_details_cached_obj[run_number_string_key] = osiris_algs.get_run_details(
+            run_number_string=run_number_string, inst_settings=self._inst_settings, is_vanadium_run=self._is_vanadium)
+
+        return self._run_details_cached_obj[run_number_string_key]
+
+    def _output_focused_ws(self, processed_spectra, run_details):
+        """
+        Takes a list of focused workspace banks and saves them out in an instrument appropriate format.
+        :param processed_spectra: The list of workspace banks to save out
+        :param run_details: The run details associated with this run
+        :param output_mode: Optional - Sets additional saving/grouping behaviour depending on the instrument
+        :return: d-spacing group of the processed output workspaces
+        """
+        d_spacing_group, tof_group, q_squared_group = common_output.split_into_tof_d_spacing_q_squared_groups(
+            run_details=run_details, processed_spectra=processed_spectra)
+        save_data(workspace_group=d_spacing_group,
+                  output_paths=self._generate_out_file_paths(run_details=run_details, unit="_d_spacing"))
+        save_data(workspace_group=tof_group,
+                  output_paths=self._generate_out_file_paths(run_details=run_details, unit="_tof"))
+        save_data(workspace_group=q_squared_group,
+                  output_paths=self._generate_out_file_paths(run_details=run_details, unit="_q_squared"))
+
+        return d_spacing_group, tof_group
+
+    def _generate_out_file_paths(self, run_details, unit=""):
+        """
+        Generates the various output paths and file names to be used during saving or as workspace names
+        :param run_details: The run details associated with this run
+        :return: A dictionary containing the various output paths and generated output name
+        """
+        output_directory = os.path.join(self._output_dir, run_details.label, self._user_name)
+        output_directory = os.path.abspath(os.path.expanduser(output_directory))
+        dat_files_directory = output_directory
+        if self._inst_settings.dat_files_directory:
+            dat_files_directory = os.path.join(output_directory,
+                                               self._inst_settings.dat_files_directory)
+
+        file_type = "" if run_details.file_extension is None else run_details.file_extension.lstrip(
+            ".")
+        out_file_names = {"output_folder": output_directory}
+        format_options = {
+            "inst": self._inst_prefix,
+            "instlow": self._inst_prefix.lower(),
```

Comment:
do we need to keep this naming convention? Its a little tricky to read

Reply:
It is the standard in the powder scripts. I am not sure If we want to change it.